### PR TITLE
Camera

### PR DESCRIPTION
**Pull request by CrAzYPiLoT-SS13**
Updated the 60's style camera, removed film and added SD cards. In
future updates I will add SD card readers and photo printers.

### CODE REVIEW

**Comment by Nienhaus** [comment]
Why not add a second type of camera?

**Comment by tigercat2000** [comment]
Um. Why? If you ask me, the 60's style looks wonderful

**Comment by Fox-McCloud** [comment]
:-1: 

SS13 is intentionally retro in a lot of ways---it's part of the theme; it's why things like tape recorders, pixel-animated pAIs,and command-line terminals are still a thing.

**Comment by Spacemanspark** [comment]
We don't have command line consoles in Paradise. That's Goon.
Personally, I think this has potential. Maybe laptops can look through the photos when an SD card is inserted.
That said, we should also keep the old cameras as well.

**Comment by CrAzYPiLoT-SS13** [comment]
I got that idea too while I was out today and I will make a change that will allow old cameras to exist too. But where should we put them? Also tiger, I like them too, but we are in space, man! Maybe they can use digital cameras to do research and security and journalists and the librarian could use the old one. Spaceman, I will add that feature, it's on my to-do list, but it won't be easy.

**Comment by tigercat2000** [comment]
I'd be okay with that, hi-tech science cameras sound amusing

File: code/modules/paperwork/photography.dm [hunk 1]
```diff
@@ -174,9 +174,9 @@
 /obj/item/device/camera/attackby(obj/item/I as obj, mob/user as mob, params)
 	if(istype(I, /obj/item/device/camera_film))
 		if(pictures_left)
-			user << "<span class='notice'>[src] still has some film in it!</span>"
+			user << "<span class='notice'>[src] still has some space left in it's card!</span>"
 			return
-		user << "<span class='notice'>You insert [I] into [src].</span>"
+		user << "<span class='notice'>You insert the [I] into [src].</span>"
```
Comment (tigercat2000) [inline code/modules/paperwork/photography.dm]: should be '\the', incase it's a named object
ex "You insert the Favorite Moments into the camera"

**Comment by CrAzYPiLoT-SS13** [comment]
You mean like, it check if it is named, if it is, it adds a 'the' prefix?

**Comment by tigercat2000** [comment]
as far as I can tell, it only checks for the name of the object being capitalized.

**Comment by CrAzYPiLoT-SS13** [comment]
Well, SD Card is already capitalized so that could cause problems.

**Comment by tigercat2000** [comment]
aahh.. up to you I guess, items are just generally named lowercase unless changed

**Comment by CrAzYPiLoT-SS13** [comment]
Hm, I don't know. I will deal with this after I add back the old ones.

**Comment by CrAzYPiLoT-SS13** [comment]
Question: do we use mapmerge?

**Comment by tigercat2000** [comment]
we do

**Comment by CrAzYPiLoT-SS13** [comment]
There we go. Also, if the sounds don't work, tell me and I'll convert them to ogg. Weird, if you ask me.

**Comment by tigercat2000** [comment]
as a general rule with byond, always use OGG
byond tends to go absolute insane with resource usage playing anything else

**Comment by CrAzYPiLoT-SS13** [comment]
Aw crap, lemme fix this.

**Comment by CrAzYPiLoT-SS13** [comment]
Here. BYOND + technology = shit.

**Comment by CrAzYPiLoT-SS13** [comment]
Aw sorry, didn't see that I commited that too -.-

**Comment by CrAzYPiLoT-SS13** [comment]
This is all.

File: code/modules/paperwork/photography.dm [hunk 2]
```diff
@@ -256,10 +271,10 @@
 	if(!on || !pictures_left || ismob(target.loc)) return
 	captureimage(target, user, flag)
 
-	playsound(loc, pick('sound/items/polaroid1.ogg', 'sound/items/polaroid2.ogg'), 75, 1, -3)
+	playsound(loc, 'sound/items/camera_shutter.wav', 75, 1, -3)
```
Comment (tigercat2000) [inline code/modules/paperwork/photography.dm]: .wav, file doesn't exist either

**Comment by tigercat2000** [comment]
@CrAzYPiLoT-SS13 Heads up, your travis build failed.

**Comment by CrAzYPiLoT-SS13** [comment]
Aw god, anything else?

**Comment by tigercat2000** [comment]
well optimally you should really probably make the polaroid a subtype of camera, so you don't have a file twice as big, and just make it change the sounds and call the parent

**Comment by CrAzYPiLoT-SS13** [comment]
Nah, too much work. You can do it if you want it though.

**Comment by tigercat2000** [comment]
Just sayin, the PR isn't likely to get merged with a crazily copied file

**Comment by CrAzYPiLoT-SS13** [comment]
If it doesn't get merged I will personally slit everyone's throats down at CC, I mean, I will make the thing you suggested.

**Comment by TheDZD** [comment]
Yeaaaah, the new camera type should be a subtype of the current one, it means you don't have to redefine all the procs for it. In general, copypasta = bad, OOP = good.

**Comment by CrAzYPiLoT-SS13** [comment]
Well than, I'll get on that tomorrow. Cough cough code faggots
...JK

**Comment by SamCroswell** [comment]
:-1:
We don't need two kinds of cameras, in my opinion. The current one works just fine, and fits SS13's current theme.

In addition, pictures of the new icons would be much appreciated.

**Comment by CrAzYPiLoT-SS13** [comment]
Space stations and 60s don't work well in my opinion. But I think the polaroids are good for civvies. The digital cameras will be science and sec only.

**Comment by CrAzYPiLoT-SS13** [comment]
There you have it gentlemen!

**Comment by ZomgPonies** [comment]
This code is extremely shoddy. Your SD cards won't work, period. There's no checks that makes sure SD cards go in a regular camera and film in the old camera, etc.

**Comment by CrAzYPiLoT-SS13** [comment]
They will work, that is tested, and I am working on the other problem now. I didn't say you have to merge it immediately.

**Comment by Nienhaus** [comment]
Those sprites aren't that great.

**Comment by CrAzYPiLoT-SS13** [comment]
I know, but I am not skilled enough. If you can do better, please do.

**Comment by Nienhaus** [comment]
I have a couple of other projects I'm supposed to be doing, if I happen to get around to it, I'll link it in here.

**Comment by CrAzYPiLoT-SS13** [comment]
Thanks

**Comment by Fox-McCloud** [comment]
Space Station and 60's don't work, but the 80's do--and that's generally what SS13's theme is based around in terms of woo-wow ness of technology and such.

**Comment by CrAzYPiLoT-SS13** [comment]
Well than, I agree with that, but would science go around carrying some shitty polaroid from 1980s in the year 2558? We are keeping the old polaroids, but science deserves something more hi-tech. Oh, and I have updated the sprites.

**Comment by CrAzYPiLoT-SS13** [comment]
I am just quick-testing the new restriction system, and if it works, I'll commit. If it doesn't, I'm gonna do it tomorrow.

**Comment by Fox-McCloud** [comment]
still don't think the sprites look that great, TBH.

Also, the term "SD Card" and "SD" on any of the sprites? It's a registered Trademark; you can dance around it and name it something very similar, but not SD or use something of its likeness.

**Comment by CrAzYPiLoT-SS13** [comment]
Is it? Didn't know. Suggestions?

**Comment by Fox-McCloud** [comment]
http://en.wikipedia.org/wiki/Secure_Digital

Yes, they use the logo+name to enforce compliance with their standards.

**Comment by CrAzYPiLoT-SS13** [comment]
Hm, how about DSS? Digitally Secure Storage?

**Comment by CrAzYPiLoT-SS13** [comment]
There you have it, subtypes, Digitally Secure Storage cards and two types
of cameras. I'm gonna start working on the photo printers now, but the
laptop view picture will be tricky. I don't even know what to do about it.

**Comment by CrAzYPiLoT-SS13** [comment]
Oh fuck forgot the sounds, minute please

**Comment by CrAzYPiLoT-SS13** [comment]
Anything new on this? BTW, the code for saving pictures and spawning them when inserted into a photo printer is tough as hell.

**Comment by Fox-McCloud** [comment]
still a :-1: from me.

**Comment by CrAzYPiLoT-SS13** [comment]
That is your opinion, but I just want you to think about the lore. Year 2558, polaroid cameras, really? The space era. And we still use manual focus wheels.

**Comment by Spacemanspark** [comment]
I still say :+1: .
Take it to le forums.

**Comment by ZomgPonies** [comment]
Same :-1: from me also. Shoddy code, shoddy sprites, useless addition that is literally identical in gameplay to the already existing camera. It doesn't hold up to the minimum code quality we require from public contributions unfortunately.

**Comment by CrAzYPiLoT-SS13** [comment]
Do not merge just yet. I found a big bug that needs to be solved and I'll
introduce changes to save into card and not spawn a photo immediately.

**Comment by CrAzYPiLoT-SS13** [comment]
Zomg, did you even read the description? This is just the start. Photo
printers, laptop picture viewers. And how is exactly this shoddy code? I
just took the existing camera, made it a subtype, and added in new sounds
and sprites. If it is shoddy code, blame it on Exadv

**Comment by ZomgPonies** [comment]
1. Your code didn't compile
2. I warned you about an easy, obvious issue regarding SD cards and you dismissed it out of hand without investigating. If you can't be bothered to correct issues maintainers report to you, we can't be bothered to merge your PRs.
3. All this PR was is an unneeded second sprite for cameras. Come up with a unique, orginal feature that brings something to the table, finish it, THEN submit a PR for review.

**Comment by CrAzYPiLoT-SS13** [comment]
1. It didn't compile because it's not done. I need someone to put do not
merge on this.
2. I fixed it. I don't know how you got to the conclusion that I didn't.
3. Well, that's all on me. I should have put a WIP tag on it. Sorry. This
just adds a dose of reality, just like the D&D figures add some fun to the
game.

**Comment by ZomgPonies** [comment]
2. What you did is copypaste code. That's not a fix. That's shoddy coding.

As I've tried to explain, this PR doesn't meet the minimum quality requirements we expect from public contributions. Fix it up, optimize it, propose something unique and useful and complete the feature, THEN submit a pull request and we'll go from there.

**Comment by Fox-McCloud** [comment]
As just a general side---your avatar choice isn't exactly a strong opinion swayer, either.

**Comment by CrAzYPiLoT-SS13** [comment]
If it really is as you say, I am sorry.
I am optimizing it.
Fox, I thought it was funny. D:

**Pull request closed without merging**